gobblin-api/src/main/java/gobblin/source/extractor/Watermark.java

```diff
@@ -0,0 +1,11 @@
+package gobblin.source.extractor;
+
+import gobblin.fork.Copyable;
+
+import org.codehaus.jackson.annotate.JsonTypeInfo;
+
+@JsonTypeInfo(use=JsonTypeInfo.Id.CLASS, include=JsonTypeInfo.As.PROPERTY, property="@class")
+public interface Watermark extends Comparable<Watermark>, Copyable<Watermark> {
```

Comment:
There's already a Watermark interface in `gobblin.source.extractor.watermark`. Better to name this one `ComplexWatermark` or something like that.